deploy/helm/edot-collector/kube-stack/values.yaml

```diff
@@ -204,344 +181,50 @@ collectors:
-                - key: transaction.root
-                - key: transaction.type
-                - key: metricset.name
-                  default_value: service_transaction
-                - key: elasticsearch.mapping.hints
-                  default_value: [aggregate_metric_double]
-              conditions:
-                - attributes["event.outcome"] != nil and attributes["event.outcome"] == "success"
-              unit: us
-              histogram:
-                buckets: [1]
-                count: Int(AdjustedCount())
-                value: Int(AdjustedCount())
-            - name: event.success_count
-              description: Success count as a metric for service transaction
-              include_resource_attributes:
-                - key: service.name
-                - key: deployment.environment # service.environment
-                - key: telemetry.sdk.language # service.language.name
-                - key: agent.name # set via elastictraceprocessor
-              attributes:
-                - key: transaction.root
-                - key: transaction.type
-                - key: metricset.name
-                  default_value: service_transaction
-                - key: elasticsearch.mapping.hints
-                  default_value: [aggregate_metric_double]
-              conditions:
-                - attributes["event.outcome"] != nil and attributes["event.outcome"] != "success"
-              unit: us
-              histogram:
-                buckets: [0]
-                count: Int(AdjustedCount())
-                value: Double(0)
       exporters:
         # [Debug exporter](https://github.com/open-telemetry/opentelemetry-collector/blob/main/exporter/debugexporter/README.md)
         debug:
           verbosity: basic
-        # [Elasticsearch exporter](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/elasticsearchexporter/README.md)
-        elasticsearch/otel:
-          endpoints:
-            - ${env:ELASTIC_ENDPOINT}
-          api_key: ${env:ELASTIC_API_KEY}
-          metrics_dynamic_index:
-            enabled: true
-          logs_dynamic_index:
-            enabled: true
-          traces_dynamic_index:
-            enabled: true
-          flush:
-            interval: 10s
-          # tls:
-          #   insecure_skip_verify: true
-          mapping:
-            mode: otel
-        # [Elasticsearch exporter](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/elasticsearchexporter/README.md)
-        elasticsearch/ecs:
-          endpoints:
-            - ${env:ELASTIC_ENDPOINT}
-          api_key: ${env:ELASTIC_API_KEY}
-          # tls:
-          #   insecure_skip_verify: true
-          mapping:
-            mode: ecs
+        # [Loadbalancing exporter](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/loadbalancingexporter)
+        loadbalancing/gateway:
+          routing_key: service
```

Comment:
I suppose you're implementing the load balancing in order to help with spreading the load for the metric aggregations. However, I'm not sure if load balancing on service name is doing more harm or good. The risk is that this could lead to hot-spotting specific instances if there's a service that's a lot more chatty than others. For APM Server, we've just been doing a random load-balancing so far.

I'd like to get opinions from @axw @simitt @lahsivjar on what you think we should start with for routing OTLP traffic from daemonsets to a set of gateway collectors.

I think that for very large-scale deployments, we need something more akin to what we're working on for HOTel, and that will probably be managed via a separate operator.

Reply:
I would also be inclined to keep it randomly distributed, unless there's a need for some stateful processing that requires it to be partitioned.

Reply:
Hi all 👋 Just my 2 cents here! Based on my experience, nodes within the same zone as the requester often tend to be favored. Thus, I would also vote starting with a randomly distributed approach as it seems like the safer option initially and it helps avoid potential hot-spotting issues 🙂

Reply:
Thanks for the feedback, looks like there's an agreement on using random distribution for now. Removed the service based approach to use the default (traceID based) https://github.com/elastic/elastic-agent/pull/6444/commits/30512425f40de321ca9e6ceeeead569e35bf3339

Reply:
Resolving the thread as the `loadbalancing` was removed in favor of a headless service otlp endpoint.